src/gridcoin/voting/registry.cpp

```diff
@@ -303,12 +309,276 @@ int64_t PollReference::Age(const int64_t now) const
 
 bool PollReference::Expired(const int64_t now) const
 {
-    return Age(now) > m_duration_days * 86400;
+    // The casting is done to suppress the comparison of signed vs unsigned. The multiplication will not
+    // overflow until 68 years, which is longer than the longest possible poll.
+    return Age(now) > (int64_t) (m_duration_days * 86400);
 }
 
 int64_t PollReference::Expiration() const
 {
-    return m_timestamp + (m_duration_days * 86400);
+    // The casting is done to suppress the comparison of signed vs unsigned. The multiplication will not
+    // overflow until 68 years, which is longer than the longest possible poll.
+    return m_timestamp + (int64_t) (m_duration_days * 86400);
+}
+
+CBlockIndex* PollReference::GetStartingBlockIndexPtr() const
+{
+    uint256 block_hash;
+    CTransaction tx;
+
+    GetTransaction(*m_ptxid, tx, block_hash);
+
+    if (!block_hash.IsNull()) {
+        CBlockIndex* pindex = mapBlockIndex[block_hash];
```

Comment:
This one (and `GetEndingBlockIndexPtr`) can just be `return mapBlockIndex[block_hash];`